sdk/azcore/etag.go

```diff
@@ -0,0 +1,80 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"strings"
+)
+
+// ETag is a property used for optimistic concurrency during updates
+// ETag is a validator based on https://tools.ietf.org/html/rfc7232#section-2.3.2
+type ETag struct {
+	value *string
+}
+
+// NewETag creates a new ETag struct
+func NewETag(value string) *ETag {
+	return &ETag{value: &value}
+}
+
+// ComparisonType specifies what type of comparison to use, Strong or Weak
+type ComparisonType string
+
+const (
+	Strong ComparisonType = "strong"
+	Weak   ComparisonType = "weak"
+)
+
+// Equals determines whether two entity-tags are equal. There are two types of comparison: Strong and Weak.
+// Strong comparison: two ETags are equivalent if both are not weak and their opaque-tags match character-by-character
+// Weak Comparison: two ETags are equivalent if their opaque-tags match character-by-character regardless of either or both being tagged as "weak"
+func (e ETag) Equals(right ETag, comparisonKind ComparisonType) bool {
+	// ETags are != if one value is null
```

Comment:
nil; not null

Reply:
I'm not sure what is happening here. Maybe you don't mean null/nil as a string in Go is never nil. A pointer to a string can be nil but you're not checking the pointer. Maybe you mean empty string instead of null/nil?
I'm not following the logic from the comments.

Reply:
I fixed this to add the logic that two `nil` `ETag`s will be equal, and they will be unequal if only one is `nil`